neonvm-runner/cmd/main.go

```diff
@@ -487,17 +488,42 @@ func runQEMU(
 
 	callbacks = cpuServerCallbacks{
 		get: func(logger *zap.Logger) (*vmv1.MilliCPU, error) {
-			return lo.ToPtr(vmv1.MilliCPU(lastValue.Load())), nil
+			switch cfg.cpuScalingMode {
+			case vmv1.CpuScalingModeSysfs:
+				cpu, err := getNeonvmDaemonCPU()
+				if err != nil {
+					logger.Error("failed to get CPU from NeonVM Daemon", zap.Error(err))
+					return nil, err
+				}
+				storedCpu := vmv1.MilliCPU(lastValue.Load())
+				if storedCpu.RoundedUp() != cpu.RoundedUp() {
+					logger.Warn("CPU from NeonVM Daemon does not match stored value, returning daemon value to let controller reconcile correct state", zap.Any("stored", storedCpu), zap.Any("current", cpu))
+					return &cpu, nil
+				}
+				logger.Debug("Returning stored CPU value to controller", zap.Any("cpu", storedCpu))
```

Comment:
question: Do we output debug logs? If no, is it worth keeping this?

Reply:
I think it is wort having it for a potential debugs, right no there is no need, but I think is easier to have it logged at debug level and enable debug level if it is required, comparing to the case that if we need it, we need to add logging, cut release and redeploy. Let me know, if you don't like the approach, I can drop it.

Reply:
I agree with Misha, I think even if we can't enable it now, but it still serves as documentation, and in case we will have an ability to enable debug logs.

Reply:
Ack, fair enough - resolving

Reply:
dropped